target/linux/tegra/base-files/lib/tegra.sh

```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
```

Comment:
> Yes there is valid model in /tmp/sysinfo/model, but during creation of image the metadata: board is set to tegra.

The image metadata boardname is validated against  /tmp/sysinfo/board_name, which is the compatible string from the dts. If you want to support multiple boards with one image you need to add all of them (space separated) to the `SUPPORTED_DEVICES`.

Either I get something wrong or it never worked as intended.

> When I did sysupgrade it failed the metadata check (it is not mandatory, so it did proceed further).

Please make it mandatory. Add `REQUIRE_IMAGE_METADATA=1` to the top of `target/linux/tegra/base-files/lib/upgrade/platform.sh`.